describe.py

```diff
@@ -402,14 +402,23 @@ def document_api(
             documentation should be saved.
         artifact_destination_dir (str): relative path where the discovery
             artifacts should be saved.
+        discovery_uri_template (str): URI template of the API's discovery document.
+            If this parameter is set, the `uri` parameter is ignored and the uri
+            will be created from this template.
     """
-    http = build_http()
-    resp, content = http.request(
-        uri
-        or uritemplate.expand(
+    # Use the discovery_uri_template to create the uri if provided
+    if discovery_uri_template:
+        uri = uritemplate.expand(
+            discovery_uri_template, {"api": name, "apiVersion": version}
+        )
+
+    if not uri:
+        uritemplate.expand(
             FLAGS.discovery_uri_template, {"api": name, "apiVersion": version}
```

Comment:
This seems like an anti-pattern (depending on a global) that can be cleaned up in the future.

Reply:
The order for constructing `uri` is use `discovery_uri_template` if provided, then use `uri` argument if provided, then fallback to `FLAGS.discovery_uri_template`

It's not clear how this can be cleaned up to preserve this order

Reply:
We could instead do something like this:

```python3

uri = (discovery_uri_template and discovery_uri_template, {"api": name, "apiVersion": version}) 
            or uri or uritemplate.expand(FLAGS.discovery_uri_template, {"api": name, "apiVersion": version})
```

Reply:
See my comment below which touches on this.

Reply:
Fixed in https://github.com/googleapis/google-api-python-client/pull/2443/commits/7a35f8c26b0452a150cac1e3b02a82fc6a1bc30b